bundles/org.openhab.binding.cbus/src/main/java/org/openhab/binding/cbus/handler/CBusCGateHandler.java

```diff
@@ -0,0 +1,511 @@
+                String commentToken = commentTokenizer.peek();
+                if (commentToken.equals("SyncUpdate")) {
+                    commentTokenizer.poll();
+                    @Nullable
+                    String address = commentTokenizer.poll();
+
+                    @Nullable
+                    String level = commentTokenizer.poll();
+                    level.replace("level=", "");
+                    updateGroup(address, level);
+                }
+            } else {
+                logger.warn("Received unparsed event: '{}'", status);
+            }
+        }
+
+    }
+
+    @NonNullByDefault
+    private class EventMonitor extends EventCallback {
+
+        @Override
+        public boolean acceptEvent(int event_code) {
+            return true;
+        }
+
+        @Override
+        public void processEvent(@Nullable CGateSession cgate_session, int eventCode,
+                @Nullable GregorianCalendar event_time, @Nullable String event) {
+            if (event == null)
+                return;
+            LinkedList<String> tokenizer = new LinkedList<String>(Arrays.asList(event.trim().split("\\s+")));
+            // List<String> tokenizer = Collections.synchronizedList(new LinkedList<String>());//
+            // Arrays.asList(event.trim().split("\\s+"))));
+            if (eventCode == 701) {
+                @Nullable
+                String address = tokenizer.poll();
+                // String oid = tokenizer.poll();
+                @Nullable
+                String value = tokenizer.poll();
+                if (value.startsWith("level=")) {
+                    String level = value.replace("level=", "");
+                    updateGroup(address, level);
+                }
+            }
+        }
+
+    }
+
+    private void updateGroup(@Nullable String address, @Nullable String value) {
+        if (address == null || value == null)
+            return;
+        String[] addressParts = address.trim().replace("//", "").split("/");
+        int application = Integer.parseInt(addressParts[2]);
+        updateGroup(addressParts[1], application, addressParts[3], value);
+    }
+
+    private void updateGroup(String network, int application, String group, String value) {
+        for (Thing networkThing : getThing().getThings()) {
+            // Is this networkThing from the network we are looking for...
+            if (networkThing.getThingTypeUID().equals(CBusBindingConstants.BRIDGE_TYPE_NETWORK) && networkThing
+                    .getConfiguration().get(CBusBindingConstants.PROPERTY_ID).toString().equals(network)) {
+                ThingHandler thingHandler = networkThing.getHandler();
+                if (thingHandler == null) {
+                    continue;
+                }
+                // Loop through all the things on this network and see if they match the application / group
+                for (Thing thing : ((CBusNetworkHandler) thingHandler).getThing().getThings()) {
```

Comment:
It would be much cleaner if you put all the channel update logic into their relevant handlers. Doing that would make it clear what channels each handler is responsible for. It looks like a couple of your handlers don't do anything which I'm sure is not the case because their logic was probably all put into their bridges.

Reply:
I had been thinking that as well.
I inherited this project and made it work reliably
So most of this code is as it was when i took it over.
Will see where i get to over the weekend with all the feedback.
Thanks

Reply:
Made this call in to each ThingHandler for that class to decide if it should handle it and to do the work.